src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventLogger.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class DiagnosticEventLogger : ILogger
+    {
+        private const string ErrorCode = "errorCode";
+        private const string HelpLink = "helpLink";
+        private readonly IDiagnosticEventRepository _diagnosticEventRepository;
+
+        public DiagnosticEventLogger(IDiagnosticEventRepository actionableEventRepository)
+        {
+            _diagnosticEventRepository = actionableEventRepository;
+        }
+
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            return null;
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return true;
+        }
+
+        private bool IsDiagnosticEvent(IDictionary<string, object> state)
+        {
+            return state.Keys.Contains("diagnosticEvent") && state.Keys.Contains("helpLink");
```

Comment:
let's make these keys more unique -- like "MS_DiagnosticEvent"

Reply:
In fact, I think we should drop "Diagnostic" since it's so overloaded everywhere. Can we call it something like "RichMessage"?

Reply:
Also, is HelpLink required? We'll always include it, but it may not be worth making it required. If something writes out "MS_DiagnosticEvent" (or whatever we call it), we should probably treat it as one of these events anyway.

Reply:
Going to removed the additional conditions.